### DISCUSSION TITLE
Predict on new videos using trained model

### DISCUSSION THREAD

**Discussion by D19CrMei** [talmolab/sleap#846] [Help!]
Hello, now it is only possible to label and train predictions within one video, but after training, will a weight file be generated? If I put a new video directly after training, how should SLEAP predict and output?

**Comment by roomrys** [comment]
Hi @D19CrMei,

### Training Pipeline (Video added before training)

There is an option to predict on all videos in your project. Any user-labels in the project (i.e. any user-labels in any video in your project) will be used in training.

![image](https://user-images.githubusercontent.com/38435167/179789155-279bcd32-5f38-4ddd-a5bb-4336735081ea.png)
**Fig 1:** Option to predict on all videos immediately after training finishes (located in Training Pipeline).

---

### Inference Pipeline (Video added after training)

If you added a new video after you have trained then you will need to navigate to the Inference Pipeline (Predict > Run Inference) to run inference on the new video.

![image](https://user-images.githubusercontent.com/38435167/179790030-770a3a12-c390-463b-b50c-3c8ed8aa2219.png)
**Fig 2:** Option to predict on all videos using a trained model (located in Inference Pipeline).

Thanks,
Liezl

**Comment by D19CrMei** [comment]
Can it only be done in the gui? Is there a way to predict new videos directly using the command line? thanks

**Reply by roomrys** [reply]
Hi @D19CrMei,

To predict on new videos using the command line, you can use the following bash script (originally [described here](https://github.com/talmolab/sleap/issues/668#issuecomment-1064305053)):
```bash
for video in path/to/videos/*.mp4
do
  sleap-track -m "my/model" "$video"
done
```

Thanks,
Liezl
